main.tf

```diff
@@ -227,6 +238,16 @@ data "aws_iam_policy_document" "default" {
   }
 }
 
+resource "aws_s3_bucket_notification" "bucket_notification_replication" {
+  count  = var.replication_enabled && var.notification_enabled ? 1 : 0
```

Comment:
you need to sort out your conditional inside the count like in the default bucket notification

Reply:
This has been amended but needs a further change